### ISSUE TITLE
Provide pre-configured Webserver Environment for use with Install Profile

### ISSUE THREAD

**Issue by patrickzurek** [eXtensibleCatalog/Drupal-Toolkit#111]
JIRA issue created by: rcook
Originally opened: 2012-07-30 11:59 AM

Issue body:

In response to some recent discussion on the XC General List about the high bar that currently exists for the Drupal Toolkit use, we had a developer discussion today about ways to approach.   A number of things were discussed regarding ways to make the installation both be easier and successful.  The first thing we will do is address some of the prerequisite hurdles by using XAMP.

I invite the developers to add more substantial content to this issue.

Other things discussed include a package management approach.  I will open a separate issue for this.

[~mwesley] [~rcook] [~bgant] [~cdelis]

**Comment by patrickzurek** [comment]
JIRA Comment by user: rcook
JIRA Timestamp: 2012-07-30 12:04 PM

Comment body:

Notify [~rcook]

**Comment by patrickzurek** [comment]
JIRA Comment by user: admin
JIRA Timestamp: 2012-07-30 03:36 PM

Comment body:

One potential issue I see with XAMPP is the latest version installs PHP 5.4, which is not compatible with the current version of Drupal that the Drupal Toolkit uses. Version 1.7.7 of XAMPP installs PHP 5.3, though you have to seek it out:

http://sourceforge.net/projects/xampp/files/XAMPP%20Linux/1.7.7/

**Comment by patrickzurek** [comment]
JIRA Comment by user: mwesley
JIRA Timestamp: 2012-08-06 11:29 AM

Comment body:

After a few days of testing, there are other issues with XAMPP. First, there is the Drupal Toolkit requirement that Java be installed. This is a not part of XAMPP, so the person setting up the system would have to first ensure it was installed, or Solr would not work. Second, XAMPP does not come with certain required features of PHP, such as PHP cURL. Without this, harvesting and NCIP (availability, authentication, etc.) would not work. 

The system requirements are a bit too complex for a point and click install, unless we go as far as to provide a system image. So, now I would like to focus on creating an Ubuntu/Debian package. Then, perhaps create other packages for other distributions. With that, setting up the Drupal Toolkit would be as simple as (1) adding our source repository to their list of package sources and (2) doing something on the command line like: "sudo apt-get install xc-drupal-toolkit".

I have somewhat limited experience with creating .deb packages. Brandon can you help with this?

**Comment by patrickzurek** [comment]
JIRA Comment by user: rcook
JIRA Timestamp: 2012-08-20 01:50 PM

Comment body:

[~sranganathan] [~bgant]

Few questions and items:
1.  I would like us to announce this new easier install process when it is ready.  We need to be clear what it is intended for, the audience and purpose (e.g. not production system, etc.)    What needs to happen before we can make that announcement?  MT, can you draft it?
2.  Do you all think there is value in  doing an Ubuntu version?
3. Regarding production use (linux/unix), what can be done to make that easier?  We at least need to spell out the requirements/expectations.  Brandon had suggested Debian?  Can we/should we pursue that?

**Comment by patrickzurek** [comment]
JIRA Comment by user: rcook
JIRA Timestamp: 2012-08-21 12:21 PM

Comment body:

The current course of action on this topic is:
1.  Sharmila to update our website to simplify and clarify the instructions and to get her Windows XAMPP instructions loaded.  After this we will support two install proccess.  The first will be a Windows based install process that is quick and fairly painless for insitutions will limited IT staff and skills to try out the DT.  The second will be the current install process that requires a broader set of skills to build and adminsiter a web server.  Issue Drupal-486
2.  Ralph will then create an Ubuntu server for Sharmila, and then she will work out and write the instructions needed to have success with an Ubuntu XAMPP scenario.   Ralph to provide input on what he did (if there is anything "special", etc.).  Issue Drupal-487
3.  Ralph will investigate and hopefully build a VM image that can be deployed easily.   We will go this route first, before exploring Debian.  I will move this issue to Ralph to reflect that work, and items 1 and 2 above will have separate Jira numbers.

**Comment by patrickzurek** [comment]
JIRA Comment by user: Brandon Gant (bgant)
JIRA Timestamp: 2012-08-22 11:39 AM

Comment body:

I'm a SysAdmin who spends his time typing "sudo apt-get install" on Ubuntu or "sudo yum install" on CentOS to setup servers. I built a simple Ubuntu .deb package a while back, but don't remember much about it other than it was pretty easy. The Ubuntu Packaging/Developers Guide would be the place to start: http://developer.ubuntu.com/packaging/html/

I standardized CARLI on Ubuntu Server for two reasons: 
1) Clear/Consistent Release Schedule: If you setup Ubuntu Server 10.04 LTS, then you can be pretty confident that the developers will not install any big software upgrades that will break anything over the five years it is supported. "10.04" means "2010 April" and LTS means "Long Term Support" (https://wiki.ubuntu.com/LTS).
2) Ubuntu is Debian based: I started out on Red Hat and it always felt clunky to me. I have been impressed with Ubuntu ever since I gave it a try. We still have a few applications that require Red Hat, so we use free CentOS for those instead of Ubuntu.

To get the developers in sync, I think we need to pick a baseline operating system that we know the software works on. Then we can branch out and create ports to other operating systems (and deal with the differences between operating systems). My current development environment is Ubuntu Server 10.04 LTS 64-bit, but we can standardize on anything we want.

**Comment by patrickzurek** [comment]
JIRA Comment by user: admin
JIRA Timestamp: 2012-08-27 11:07 AM

Comment body:

For #3, we need to install VMware Studio. This is a free product, which Jan will install this week into our VMware environment. I've never used it before, so there's going to be a bit of a learning curve. Hopefully, I'll be able to produce something towards the end of next week. If it's going to take longer, I'll let everyone know.

**Comment by patrickzurek** [comment]
JIRA Comment by user: rcook
JIRA Timestamp: 2012-08-27 01:50 PM

Comment body:

Ralph,  I gave an update on recent discussions on this topic at the XC meeting today.  I came to talk with you, but you are in a long meeting to I am writing this now, be we can chat if anything us unclear.

We would like to expand on the VMWare image a bit (or Ubuntu if that route is taken), and wonder if it can now includes
- server that includes an OAI Toolkit, MST, and Drupal Toolkit and a sample file of MARC records that can be processed
- instructions to install the image
- instructions on how to setup and run the sample MARC records through the entire system
- server requirements, etc.
- configured to use nothing that is proprietary.  For example, we might be doing cover images in Drupal currently using our subscriptions service, be I think there is a way to do it by using Amazon cover images instead (MT can help with that)

The idea behind this is to provide the complete end to end system(not including NCIP yet, but maybe later)in an easy to install package that gives them the full feel of what is invovled.   We feel that perhaps just giving a Drupal Toolkit instance tha pulls from a static MST doesn't perhaps go far enough.

**Comment by patrickzurek** [comment]
JIRA Comment by user: rcook
JIRA Timestamp: 2012-10-16 03:10 PM

Comment body:

Ralph, how goes this?

**Comment by patrickzurek** [comment]
JIRA Comment by user: admin
JIRA Timestamp: 2012-10-16 03:36 PM

Comment body:

I've had to switch gears the last couple of days to work on other projects. However, I plan on spending a good chunk of Wednesday and Thursday working on this. My hope is that by the end of the week I'll have some sample records processed through all three toolkits and a draft of directions on how to do it.

**Comment by patrickzurek** [comment]
JIRA Comment by user: admin
JIRA Timestamp: 2012-10-18 04:39 PM

Comment body:

I have run into a wall with this OAI Toolkit issue:
[Fatal Error] :1:129: Attribute "xmlns" bound to namespace "http://www.w3.org/2000/xmlns/" was already specified for element "collection".
org.marc4j.MarcException: Unable to parse input
at org.marc4j.MarcXmlParser.parse(MarcXmlParser.java:95)
at org.marc4j.MarcXmlParser.parse(MarcXmlParser.java:64)
at org.marc4j.MarcXmlParserThread.run(MarcXmlParserThread.java:115)
Caused by: org.xml.sax.SAXParseException: Attribute "xmlns" bound to namespace "http://www.w3.org/2000/xmlns/" was already specified for element "collection".
at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)
at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
at org.marc4j.MarcXmlParser.parse(MarcXmlParser.java:93)
... 2 more

That error occurs during the load process (the convert seems to work normally). The version of the OAI Toolkit doesn't seem to matter...it happens with version 1.0.9 and 1.0.10. Because this environment is different (different OS, different Tomcat, different Java) it's going to be more work to troubleshoot. I still have things to try to get this to work, but it might take a while. 

Ralph

**Comment by patrickzurek** [comment]
JIRA Comment by user: rcook
JIRA Timestamp: 2012-10-18 04:58 PM

Comment body:

[~cdelis] anything you can do would be great.  This is a critical thing to get out......really critical.   We need to supply a way for folks to have success with xc.

**Comment by patrickzurek** [comment]
JIRA Comment by user: rcook
JIRA Timestamp: 2012-10-22 11:00 AM

Comment body:

[~admin]   It might be helpful if you can supply [~cdelis] with log in info so that you can perhaps collaborate a bit more on this issue.

**Comment by patrickzurek** [comment]
JIRA Comment by user: admin
JIRA Timestamp: 2012-10-22 11:53 AM

Comment body:

I figured it out. It was the Java package that I selected when putting this appliance together. Note- when you start building the appliance, you select which software packages you want installed from a list provided. If it's not in the list (like XC software) then you have an upload section where you can include it in the appliance. You then have to have it installed via a script if you want to automate it, which is no big deal. The Java package on their list appeared to be fine, but I guess the OAIToolkit didn't like it. Anyway, I'll modify the appliance to include the Java I installed. Now to test the MST.
